src/python/pants/backend/python/target_types.py

```diff
@@ -384,3 +381,15 @@ class PythonRequirementsFile(Target):
 
     alias = "_python_requirements_file"
     core_fields = (*COMMON_TARGET_FIELDS, PythonRequirementsFileSources)
+
+
+# -----------------------------------------------------------------------------------------------
+# `python_distribution` target
+# -----------------------------------------------------------------------------------------------
+
+
+class PythonDistribution(Target):
+    """A publishable python distribution."""
+
+    alias = "python_distribution"
+    core_fields = (*COMMON_PYTHON_FIELDS, PythonProvidesField)
```

Comment:
Ack that it makes sense for this target to have a `compatibility` field, right?

Reply:
It should potentially imitate binary in that regard and have `platforms`:
```
    platforms
        type: str | Iterable[str] | None, default: None
        The platforms the built PEX should be compatible with. This defaults to the
        current platform, but can be overridden to different platforms. You can give a
        list of multiple platforms to create a multiplatform PEX. To use wheels for
        specific interpreter/platform tags, you can append them to the platform with
        hyphens like: PLATFORM-IMPL-PYVER-ABI (e.g. "linux_x86_64-cp-27-cp27mu",
        "macosx_10.12_x86_64-cp-36-cp36m"). PLATFORM is the host platform e.g.
        "linux-x86_64", "macosx-10.12-x86_64", etc". IMPL is the Python implementation
        abbreviation (e.g. "cp", "pp", "jp"). PYVER is a two-digit string representing
        the python version (e.g. "27", "36"). ABI is the ABI tag (e.g. "cp36m",
        "cp27mu", "abi3", "none")
```
...

Reply:
We only use that field to figure out if we should assume py2 or. py3, and we use the compatibility string for that.  I'm not sure `platforms` are appropriate, since we wouldn't be using them anyway, other than for the interpreter version.

Reply:
I think it makes sense for this to have compatibility, but we could potentially not and rely on the values from the `python_library` targets. That might be tighter.

Reply:
> other than for the interpreter version.

Isn't that exactly what is needed here? If you're trying to build a dist for `linux_x86_64-cp-27-cp27mu`, you need an interpreter compatible with that I think. If we're unsure on this, might be worth waiting for @jsirois to review tomorrow.